code/__HELPERS/mobs.dm

```diff
@@ -474,13 +474,17 @@ GLOBAL_LIST_INIT(skin_tone_names, list(
 		. += borg
 
 //Returns a list of AI's
-/proc/active_ais(check_mind=FALSE, z = null)
+/proc/active_ais(check_mind=FALSE, z = null, skip_syndicate, only_syndicate)
 	. = list()
 	for(var/mob/living/silicon/ai/ai as anything in GLOB.ai_list)
 		if(ai.stat == DEAD)
 			continue
 		if(ai.control_disabled)
 			continue
+		if(skip_syndicate && (ROLE_SYNDICATE in ai.faction))
+			continue
+		if(only_syndicate && !(ROLE_SYNDICATE in ai.faction))
+			continue
```

Comment:
use a bitflag for this maybe, would skip the need for 2 variables
I.e.

Reply:
On second thought, this should probably not be using factions since factions are a nebulous concept more meant for AI mobs to know who and who not to attack.

Can you just separate AIs into different bitflags when they're created, i.e. AI_MODEL_SYNDICATE, AI_MODEL_NT and use these to filter out the specific silicon type for law changes, etc.

Otherwise this could lead to weird behaviour where an AI who's been made into a malfunctioning AI or gains the syndie faction in some way immune to law changes despite being NT-branded before,.

Reply:
This still uses factions and I agree that it's not ideal.